### PR TITLE
fix: ng-add, some files incorrectly generated in the context of monorepo

### PR DESCRIPTION
**Pull request by fpaul-1A**
## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

**Review by vscaiceanu-1a — APPROVED** [review]

**Review by mrednic-1A — APPROVED** [review]

File: packages/@o3r/core/schematics/ng-add/project-setup/index.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export const prepareProject = (options: NgAddSchematicsSchema) => async (tree: T
     readPackageJson, removePackages, renamedPackagesV7toV8, updateImports, isMultipackagesContext, getO3rPeerDeps
   } = await import('@o3r/schematics');
   const installOtterLinter = await shouldOtterLinterBeInstalled(context);
-  const workspaceProject = tree.exists('angular.json') ? getProjectFromTree(tree, options.projectName) : undefined;
+  const workspaceProject = (tree.exists('angular.json') && options.projectName) ? getProjectFromTree(tree, options.projectName) : undefined;
```
Comment (kpanot) [inline packages/@o3r/core/schematics/ng-add/project-setup/index.ts:44]: the function `getProjectFromTree` should be deprecated, the implementation that defaults on the first of the list should be delegated to the schematic.
I would suggest to do a `options.projectName && getWorkspaceConfig(tree)?.projects?.[options.projectName] || undefined`

File: packages/@o3r/schematics/src/utility/modules.ts [hunk 2]
```diff
@@ -104,11 +104,15 @@ export function getMainFilePath(tree: Tree, context: SchematicContext) {
  * one of its modules.
  *
  * @param tree
+ * @param options @see RuleFactory.options
```
Comment (kpanot) [inline packages/@o3r/schematics/src/utility/modules.ts:107]: it is not supposed to be `{@see RuleFactory.options}` ?

Comment (fpaul-1A) [reply packages/@o3r/schematics/src/utility/modules.ts:107]: I've copy pasted what we already have everywhere else
if we want to update that everywhere this should be done in a separate PR

File: packages/@o3r/core/schematics/ng-add/project-setup/index.ts (hunk 1 above)
Comment (mrednic-1A) [reply packages/@o3r/core/schematics/ng-add/project-setup/index.ts:44]: IMO this should be a dedicated task, as **getProjectFromTree** is extensively used across the schematic scripts of updates and in generators.

File: packages/@o3r/schematics/src/rule-factories/ng-add/index.ts [hunk 3]
```diff
@@ -29,13 +32,13 @@ export function ngAddPackages(packages: string[], options?: Omit<NgAddPackageOpt
 
   const getInstalledVersion = (packageName: string) => {
     try {
-      const packageJsonAfterInstall = JSON.parse(readFileSync('./package.json', {encoding: 'utf8'}));
+      const fileSystemPackageJson = JSON.parse(readFileSync(path.join(fsWorkingDirectory, 'package.json'), {encoding: 'utf8'}));
```
Comment (matthieu-crouzet) [inline packages/@o3r/schematics/src/rule-factories/ng-add/index.ts:35]:
```suggestion
      const fileSystemPackageJson = JSON.parse(readFileSync(path.posix.join(fsWorkingDirectory, 'package.json'), {encoding: 'utf8'}));
```

Comment (fpaul-1A) [reply packages/@o3r/schematics/src/rule-factories/ng-add/index.ts:35]: I am building an actual file system path, so I need to use `path.join`

**Review by mrednic-1A — APPROVED** [review]

**Pull request merged**